tests/test_unit_command.cc

```diff
@@ -13,14 +13,108 @@ namespace sc2 {
 
     class TestAttackAttack : public TestUnitCommandTargetingPoint {
     public:
-        TestAttackAttack() {
+        TestAttackAttack() : test_units_tags_{ 0 }, test_tags_counter_{ 0 }, TAGS_SUBTESTS_QUANTITY{ 6 } {
             test_unit_type_ = UNIT_TYPEID::TERRAN_MARINE;
             test_ability_ = ABILITY_ID::ATTACK;
         }
 
+        void AdditionalTestSetup() {
+            agent_->Debug()->DebugCreateUnit(test_unit_type_, origin_pt_, agent_->Observation()->GetPlayerID(), 2);
+            target_point_ = GetPointOffsetX(origin_pt_);
+            agent_->Debug()->DebugCreateUnit(sc2::UNIT_TYPEID::ZERG_ZERGLING, GetPointOffsetX(origin_pt_, -10), 2, 1);
+            agent_->Debug()->SendDebug();
+        }
+
         void SetTestTime() override {
-            wait_game_loops_ = 50;
+            wait_game_loops_ = 150;
         }
+
+        void IssueUnitCommand(ActionInterface* act) {
+            if (test_tags_counter_ < TAGS_SUBTESTS_QUANTITY) {
+                test_units_tags_ = ConvertToTags(test_units_);
+
+                if (!orders_verified_ && ability_command_sent_) {
+                    VerifyUnitOrders(test_unit_, test_ability_);
+                }
+
+                if (ability_command_sent_) {
+                    orders_verified_ = false;
+                    ability_command_sent_ = false;
+                    ++test_tags_counter_;
+                    return;
+                }
+
+                uint32_t current_game_loop = agent_->Observation()->GetGameLoop();
+                order_on_game_loop_ = current_game_loop + 10;
+
+                switch (test_tags_counter_) {
+                    case 0:
+                        TestTagsTargetingPoint(act);
+                        break;
+                    case 1:
+                        TestTagTargetingPoint(act);
+                        break;
+                    case 2:
+                        TestTagsTargetingUnit(act);
+                        break;
+                    case 3:
+                        TestTagTargetingUnit(act);
+                        break;
+                    case 4:
+                        TestTagsNoTarget(act);
+                        break;
+                    case 5:
+                        TestTagNoTarget(act);
+                        break;
+                    default:
+                        break;
+                }
+
+                ability_command_sent_ = true;
+            }
+        }
+
+            void TestTagsTargetingPoint(ActionInterface* act) {
```

Comment:
Please, check the indentation. Seems that it is a bit extra.

Reply:
Might be also tabs or spaces.

Reply:
Yeah, I've become dependent on auto-formatting, and don't have it setup right for this repo, so we are seeing my lack of doing it myself during my learning.  LOL.  Yup, I just need to follow the braces.